app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  belongs_to :user, foreign_key: 'author_id'
+
+  def self.count_posts(user)
+    user.update(postCounter: user.posts.size)
+  end
```

Comment:
- [ ] [OPTIONAL] Please consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#count_posts` method every time a new post is saved to the database, then make the `count_posts` method private 👍 
- [ ] [OPTIONAL] Please consider implementing the above suggestion for the `Comment` and `Like` models 🙏

Reply:
Thanks for these suggestions :) I will apply these optional suggestion soon.